extensions/amp-apester-media/0.1/amp-apester-media.js

```diff
@@ -0,0 +1,232 @@
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+import {CSS} from '../../../build/amp-apester-media-0.1.css';
+import {user} from '../../../src/log';
+import {loadPromise} from '../../../src/event-helper';
+import {Layout} from '../../../src/layout';
+import {isExperimentOn} from '../../../src/experiments';
+import {removeElement} from '../../../src/dom';
+import {vsyncFor} from '../../../src/vsync';
+
+/** @const */
+const TAG = 'amp-apester-media';
+
+/**
+ * AMP Apester-media
+ */
+class AmpApesterMedia extends AMP.BaseElement {
+  /** @override */
+  preconnectCallback(onLayout) {
+    this.preconnect.url(this.displayBaseUrl_, onLayout);
+    this.preconnect.url(this.rendererBaseUrl_, onLayout);
+  }
+
+
+  /** @override */
+  isLayoutSupported(layout) {
+    return layout == Layout.FIXED_HEIGHT;
+  }
+
+  /** @override */
+  viewportCallback(inViewport) {
+    if (inViewport && !this.seen_) {
+      if (this.iframe_.contentWindow) {
+        console.log('seen');
+        this.seen_ = true;
+        this.iframe_.contentWindow./*OK*/postMessage('interaction seen', '*');
+      }
+    }
+  }
+
+  /** @override */
+  buildCallback() {
+
+    // EXPERIMENT
+    user().assert(isExperimentOn(this.win, TAG), `Enable ${TAG} experiment`);
+
+    /**
+     * @const @private {?String}
+     */
+    this.rendererBaseUrl_ = 'https://renderer.qmerce.com';
+
+    /**
+     * @const @private {?String}
+     */
+    this.displayBaseUrl_ = 'https://display.apester.com';
+
+    /**
+     * @const @private {?String}
+     */
+    this.mediaAttribute_ = user().assert(
+      (this.element.getAttribute('data-apester-media-id') ||
+      (this.random = this.element.getAttribute('data-apester-channel-token'))),
+      'Either the data-apester-media-id or the data-apester-channel-token '
+      + 'attributes must be specified for <amp-apester-media> %s',
+      this.element);
+
+    /**
+     * @private {?Element}
+     */
+    this.iframe_ = this.element.ownerDocument.createElement('iframe');
+
+    /**
+     * @private {?Promise}
+     */
+    this.iframePromise_ = null;
+
+    /**
+     * @private {?Promise}
+     */
+    this.seen_ = false;
+    this.element.classList.add('-amp-apester-container');
+  }
+
+  /** @override */
+  firstLayoutCompleted() {
+    // Do not hide placeholder
+  }
+
+  buildUrl_() {
+    const suffix = (this.random) ? '/tokens/' + this.mediaAttribute_
+    + '/interactions/random' : '/interactions/' + this.mediaAttribute_
+    + '/display';
+    return this.displayBaseUrl_ + suffix;
+  }
+
+  queryMedia_() {
+    return new Promise(function(resolve, reject) {
+      const req = new XMLHttpRequest();
```

Comment:
Unless there are special requirements here, let's switch to our XHR service, which uses (and polyfills) fetch API. Look for `xhrFor(this.win).fetchJson()` calls.